Small cleanups to make lint and TS compiler happier

jtakalai opened the pull request:
Changes mostly in data union code, in how ethers is imported

jtakalai committed:
fix: small cleanups to make lint and TS compiler happier

timoxley commented on packages/client/tsconfig.json:
```diff
@@ -2,7 +2,8 @@
         "outDir": "dist",
-        "rootDirs": ["src", "test"]
+        "rootDirs": ["src", "test"],
+        "resolveJsonModule": true
```
`resolveJsonModule` shouldn't be necessary, did you hit an issue? https://github.com/streamr-dev/network-monorepo/blob/193e6eac1e150cd7b3f1a20a7c86e814bcb12dc0/packages/dev-config/ts/tsconfig.browser.json#L22

kare replied:
It's set in @streamr/dev-config: https://github.com/streamr-dev/network-monorepo/blob/main/packages/dev-config/ts/tsconfig.browser.json#L22

jtakalai replied:
I think tests import contract jsons directly

jtakalai replied:
I can delete it, too

timoxley reviewed (approved):
Great stuff

jtakalai committed:
Merge branch 'v5' into lint-cleanup

kare commented on packages/client/tsconfig.json:
```diff
@@ -2,7 +2,8 @@
     "compilerOptions": {
         "outDir": "dist",
-        "rootDirs": ["src", "test"]
+        "rootDirs": ["src", "test"],
```
@jtakalai @timoxley @harbu @teogeb
Any reason for not using these configuration settings?
```
    "incremental": true,                              /* Enable incremental compilation */
    "composite": true,                                /* Enable constraints that allow a TypeScript project to be used with project references. */
    "tsBuildInfoFile": ".tsbuildinfo",                /* Specify the folder for .tsbuildinfo incremental compilation files. */
```
Is the `.tsbuildinfo` cacheable in CI?

kare replied:
I see `incremental` is set in the config, but it requires `composite = true`

jtakalai replied:
no idea

jtakalai committed:
fix: revert config change

I think resolveJsonModule is needed, but it might come from another config, too.
